Batch saas synchronous api 2 (#266)
speechmatics/docs: docs/speech-to-text/batch/synchronous.mdx

---

> description: 'Block for a transcript in a single request instead of polling for job status.'

Reviewer: Personal opinion, but I find something like this clearer:

```suggestion
description: Wait for a transcription job to finish synchronously via HTTP
```

> ---

Reviewer 2: Missing `title:` in the frontmatter — only `description:` is set. The style guide requires both `title` and `description` on every page (AI-readability standards). Docusaurus falls back to the H1, so the page currently renders fine, but explicit frontmatter keeps search, retrieval, and page metadata consistent across the site.

Suggest:

```yaml
---
title: Synchronous transcription
description: Block for a transcript in a single request instead of polling for job status.
---
```

Author: adding title:

Reviewer: @petemomo worth noting title actually auto-populates from the top `h1` header content, so no obligation to add it unless the tab title should differ.

Reviewer 2: Thanks, that's it.

# Synchronous transcription

> The `wait` query parameter blocks a request until the job reaches a terminal state, so you can get a transcript in one call instead of polling. It applies to job creation, job status, and transcript retrieval.

Reviewer: I think it's good to spell out the full breadth of the feature from the top for clarity. Maybe something like this:

```suggestion
The `wait` query parameter lets you block a job request until either:
- The job enters a terminal state (`done` or `rejected`)
- The specified number of seconds elapses.

This allows you to avoid polling the API for job completion. The `wait` parameter can be used for several endpoints: i.e. creating a job, fetching job details, and fetching the full transcript.
```

:::info

> `wait` is available on SaaS on Cloud only. It is not available on on-prem deployments.

Reviewer: To me this sounds like we'll never offer it, what do you think?

```suggestion
`wait` is available on SaaS on Cloud only. It is not currently available on on-prem deployments.
```

:::

`wait` takes a number of seconds and is capped server-side. If the job has not finished when `wait` elapses, the request returns the current state and you retry to keep waiting.

## Create a job and wait for the transcript

> Pass `wait` to `POST /jobs` to block until the job finishes. Add `format` to choose the embedded transcript format (`json-v2`, `txt`, or `srt`; defaults to `json-v2`).

Reviewer: ```suggestion
Pass the `wait` query parameter to `POST /jobs` to block until the job finishes. Add `format` to choose the embedded transcript format (`json-v2`, `txt`, or `srt`; defaults to `json-v2`).
```

```json
{
  "id": "a1b2c3d4e5",
  "status": "done",
  "txt": "Welcome to Speechmatics..."
}
```

For the default `format=json-v2`, the value is a nested object with the same shape as the [transcript endpoint](/api-ref/batch/get-the-transcript-for-a-transcription-job) response, not a string. The `results` array is abbreviated below:

```json
{
  "id": "uwcl3jevp3",
  "status": "done",
  "json-v2": {
    "format": "2.9",
    "job": {
      "created_at": "2026-06-23T13:33:24.230Z",
      "data_name": "example.wav",
      "duration": 10,
      "id": "uwcl3jevp3"
    },
    "metadata": {
      "created_at": "2026-06-23T13:33:24.879071Z",
      "transcription_config": { "language": "en", "model": "enhanced" },
      "type": "transcription"
    },
    "results": [
      {
        "alternatives": [
          { "confidence": 1.0, "content": "I", "language": "en", "speaker": "UU" }
        ],
        "start_time": 1.5,
        "end_time": 1.79,
        "type": "word"
      }
    ]
  }
}
```

If the job is still running when `wait` elapses, `status` is `created` and no transcript is embedded:

```json
{
  "id": "a1b2c3d4e5",
  "status": "created"
}
```

> `status` reports the outcome of the wait, which is distinct from the [job lifecycle status](/speech-to-text/batch/output#check-single-job-status):

Reviewer: I'm a bit confused about this, I thought `status` would always return the current status of the job? Or is it just in the case of `created` after wait that it may be different?

Author: status is returned if wait parameter was specified, this is so we made the api change opt in (client has provided wait). schema of the response changes only for people who has added the wait parameter

Reviewer: Ah I see, I missed that this was specifically about the `POST /jobs` case. I think it's a little confusing to say this is "distinct from the job lifecycle status". To my eyes it refers to the same thing, even if the set of values it returns is different. I'd consider removing this line and let the below table speak for itself.

```suggestion
```
